src/app/manufacturing/cncGenerator.ts

```diff
@@ -0,0 +1,281 @@
+    __svgData: Map<String, string>;
+
+    /**
+     * Default Constructor of GNCGenerator object.
+     * @param {Device} device Device object
+     * @param {*} viewManagerDelegate
+     */
+    constructor(device: Device, viewManagerDelegate: viewManager) {
+        this.__device = device;
+        this.__viewManagerDelegate = viewManagerDelegate;
+
+        this.__svgData = new Map();
+    }
+
+    /**
+     * Gets the SVG output
+     * @returns {}
+     * @memberof CNCGenerator
+     */
+    getSVGOutputs(): Map<String, string> {
+        return this.__svgData;
+    }
+
+    /**
+     * Generate the port layers
+     * @memberof CNCGenerator
+     * @returns {void}
+     */
+    generatePortLayers(): void {
+        /*
+        Step 1 - Get all the layers
+        Step 2 - Get all the ports in each of the layers
+        Step 3 - Create a manufacturing layer
+                -  Populate with the ports
+         */
+        // let components = this.__device.components;
+        const layers: Array<Layer> = this.__device.layers;
+
+        const mfglayers: Array<ManufacturingLayer> = [];
+
+        let isControl: boolean = false;
+        let isIntegrate: boolean = false;
+
+        for (const i in layers) {
+            const layer: Layer = layers[i];
+            const ports: Array<string> = [];
+
+            const features: { [index: string]: Feature } = layer.features;
+
+            if (layer.type === LogicalLayerType.CONTROL) {
+                isControl = true;
+            } else if (layer.type === LogicalLayerType.INTEGRATION) {
+                isIntegrate = true;
+            }
+
+            for (const key in features) {
+                const feature: Feature = features[key];
+                // TODO: Include fabtype check also
+                if (feature.getType() === "Port") {
+                    ports.push(key);
+                }
+            }
+
+            if (ports.length === 0) {
+                continue;
+            }
+
+            const manufacturinglayer: ManufacturingLayer = new ManufacturingLayer("ports_" + layer.name + "_" + i);
+            // console.log("manufacturing layer :", manufacturinglayer);
+
+            for (const fi in ports) {
+                const featurekey: string = ports[fi];
+                // console.log("Key:", featurekey);
+                // console.log("rendered:feature", this.__viewManagerDelegate.view.getRenderedFeature(featurekey));
+                const issuccess: boolean = manufacturinglayer.addFeature(this.__viewManagerDelegate.view.getRenderedFeature(featurekey));
+                if (!issuccess) {
+                    console.error("Could not find the feature for the corresponding id: " + featurekey);
+                }
+            }
+
+            if (isControl) {
+                manufacturinglayer.flipX();
+                isControl = false;
+            } else if (isIntegrate) {
+                //TODO: manufacturinglayer.flipX(); if on different substrate
```

Comment:
You need to check if substrate is `+1` and flip X if that's the case.
```suggestion
                //TODO: manufacturinglayer.flipX(); if on different substrate
```

Reply:
How should this work for features which can be placed on any layer (port, edge, channel)? Do I still base it off what layer they are on rather than the substrate value? I'm unsure how to implement the substrate offset usage, since it currently works based upon layer

Reply:
Also, what should integration's substrate-offset be?

Reply:
For most of them it's "0" for the merger, its "+1"